lib/modern_searchlogic/column_conditions.rb

```diff
@@ -43,8 +43,18 @@ def searchlogic_prefix_condition(prefix, &method_block)
         searchlogic_prefix_conditions[prefix] = method_block
       end
 
+      def searchlogic_extract_arel_compatible_value(value)
+        if value.respond_to?(:map) && !value.acts_like?(:string)
```

Comment:
I pulled this out of ActiveRecord to match what they're doing under the hood: https://github.com/rails/rails/blob/ceb33f84933639d3b61aac62e5e71fd087ab65ed/activerecord/lib/active_record/sanitization.rb#L164